src/coreclr/jit/morph.cpp

```diff
@@ -2353,20 +2324,7 @@ bool Compiler::fgTryMorphStructArg(CallArg* arg)
     //
     if (newArg == nullptr)
     {
-        bool isUseOfIndependentlyPromotedStruct =
-            argNode->OperIs(GT_LCL_VAR) &&
-            (lvaGetPromotionType(argNode->AsLclVarCommon()->GetLclNum()) == PROMOTION_TYPE_INDEPENDENT);
-        if (isUseOfIndependentlyPromotedStruct)
-        {
-            if (arg->AbiInfo.HasExactlyOneRegisterSegment())
-            {
-                // We already tried to use the fields above, but that failed.
-                // Here we prefer to create a copy to avoid DNER'ing the local.
-                // That turns out to be profitable mostly for single-register arguments.
-                return false;
-            }
-        }
-        else if (!argNode->TypeIs(TYP_STRUCT) && arg->AbiInfo.HasExactlyOneRegisterSegment())
+        if (!argNode->TypeIs(TYP_STRUCT) && arg->AbiInfo.HasExactlyOneRegisterSegment())
         {
```

Comment:
All independently promoted structs passed in one register are converted to `FIELD_LIST` above with this change, so this condition is always false now.